src/actions/connectionKeyPairActions.js

```diff
@@ -232,19 +264,24 @@ export const updateConnectionKeyPairs = (
       });
 
       try {
-        const newKeyPairs =
-          await generateKeyPairThreadPool(
-            mnemonic,
-            privateKey,
-            totalConnections,
-            connectionKeyPairs.length,
-            lastConnectionKeyIndex);
-        const resultConnectionKeys = connectionKeyPairs.concat(newKeyPairs);
-        await dispatch({
-          type: UPDATE_CONNECTION_KEY_PAIRS,
-          payload: resultConnectionKeys,
-        });
-        await dispatch(saveDbAction('connectionKeyPairs', { connectionKeyPairs: resultConnectionKeys }, true));
+        if (lastConnectionKeyIndex === -1) {
+          const newKeyPairs =
+            await generateKeyPairThreadPool(
+              mnemonic,
+              privateKey,
+              totalConnections,
+              connectionKeyPairs.length,
+              lastConnectionKeyIndex);
+          const resultConnectionKeys = connectionKeyPairs.concat(newKeyPairs);
+          await dispatch({
+            type: UPDATE_CONNECTION_KEY_PAIRS,
+            payload: resultConnectionKeys,
+          });
+          await dispatch(saveDbAction('connectionKeyPairs', { connectionKeyPairs: resultConnectionKeys }, true));
+          dispatch(backgroundPreKeyGeneratorAction(mnemonic, privateKey));
```

Comment:
We could move this line outside the `if` and drop the `else` part.

Reply:
I was worried if the dispatched action would affect the UI if the context of the setTimeout changed for react-native